src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -67,16 +68,29 @@ private function registerHandlers(ContainerBuilder $container)
 
         foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
             foreach ($tags as $tag) {
-                $handles = $tag['handles'] ?? $this->guessHandledClass($r = $container->getReflectionClass($container->getParameterBag()->resolveValue($container->getDefinition($serviceId)->getClass())), $serviceId);
-
-                if (!class_exists($handles)) {
-                    $messageClassLocation = isset($tag['handles']) ? 'declared in your tag attribute "handles"' : sprintf('used as argument type in method "%s::__invoke()"', $r->getName());
-
-                    throw new RuntimeException(sprintf('Invalid handler service "%s": message class "%s" %s does not exist.', $serviceId, $handles, $messageClassLocation));
+                $handles = $tag['handles'] ?? $this->guessHandledClasses($r = $container->getReflectionClass($container->getParameterBag()->resolveValue($container->getDefinition($serviceId)->getClass())), $serviceId);
+                if (empty($handles)) {
+                    throw new RuntimeException(sprintf('The handler service "%s" does not seem to handle any message.', $serviceId));
                 }
 
                 $priority = $tag['priority'] ?? 0;
-                $handlersByMessage[$handles][$priority][] = new Reference($serviceId);
+
+                foreach ($handles as $messageClass) {
+                    if (is_array($messageClass)) {
+                        $messagePriority = $messageClass[1];
+                        $messageClass = $messageClass[0];
+                    } else {
+                        $messagePriority = $priority;
+                    }
+
+                    if (!class_exists($messageClass)) {
```

Comment:
do we really need this check?
we usually don't validate DI configurations at compile time, because this adds overhead to the compilation process

Reply:
I believe it would definitely helps the DX if there is a typo in the class name or missing `use`.